docs/source/guide/rem-3-options.md

```diff
@@ -12,3 +12,153 @@ kernelspec:
+
+
+## Options for specifying the inverse confusion matrix  
+
+Mitiq provides two functions for constructing an inverse confustion matrix from user-provided information about a device's confusion matrix. We will describe each of them and then illustrate them in code with toy examples.
+
+### `generate_inverse_confusion_matrix(num_qubits, p0, p1)`
+This function embodies the simplest noise model possible, where one assumes that noise affects the measurement of each qubit independently and with the same confusion probabilities, specified by $p_0 = Pr(1|0)$, the probability $|0\rangle$ gets flipped to $|1\rangle$ when measured, and $p_1 = Pr(0|1)$, the probability $|1\rangle$ gets flipped to $|0\rangle$). The $2 \times 2$ confusion matrix $A_1$ for the $1$st qubit (and every other qubit) is then 
+
+$$
+\begin{bmatrix}
+1-p_0 & p_1 \\
+p_0 & 1-p_1
+\end{bmatrix}
+$$
+
+and the joint $2^n \times 2^n$ confusion matrix $A$ for all $n$ qubits is just $n$ copies of $A_1$ tensored together: $A = A_1 \otimes  \dots \otimes A_1 = A_1^{\otimes n}$. 
+
+
+To construct an inverse confusion matrix with `generate_inverse_confusion_matrix()` the user supplies the number of qubits and the single-qubit confusion matrix parameters $p_0$ and $p_1$. Here is an example with two qubits.
+
+```{code-cell} ipython3
+from functools import reduce
+
+import numpy as np
+from mitiq.rem import generate_inverse_confusion_matrix
+
+# Confusion matrix for qubit 1
+A1_entries = [
+    0.9, 0.2,
+    0.1, 0.8
+]
+A1 = np.array(A1_entries).reshape(2,2)
+
+# Overall 2-qubit confusion matrix (tensor two copies of A1)
+A = np.kron(A1, A1)
+
+# Generate inverse confusion matrix.
+# For this simple error model the user only has to
+# specify the single qubit flip probabilities p0 and p1 
+A_pinv = generate_inverse_confusion_matrix(2, p0=0.1, p1=0.2)
+
+print(f"Confusion matrix:\n{A}\n")
+print(f"Column-wise sums:\n{A.sum(axis=0)}\n\n")
+print(f"Inverse confusion matrix:\n{A_pinv}")
+```
+
+### `generate_tensored_inverse_confusion_matrix(num_qubits, confusion_matrices)`
+This function allows for any stochastic noise model (i.e. any model representable classically by a $2^{n} \times 2^{n}$ confusion matrix). The general form of an overall $n$-qubit confusion matrix $A$ is a factorization $A = A^{(1)} \otimes \dots \otimes A^{(k)}$ into the tensor product of $k$ smaller confusion matrices (supplied by the user in `confusion_matrices`), one for each subgroup in a partition of the $n$ qubits into $k$ smaller subgroups. The factorization encodes the assumption that there are $k$ independent/uncorrelated noise processes affecting the $k$ disjoint subgroups of qubits (possibly of different sizes), but within each subgroup noise may be correlated between qubits in that subgroup. This model includes the simplest noise model above as the special case where $k=n$ and each of the $n$ single-qubit subgroups has the same confusion matrix $A_1$:
```

Comment:
I'd recommend using "subspaces" rather than subgroups, but don't have a strong opinion about it.

Reply:
I agree that _subgroups_ can be confusing due to its mathematical meaning. 
My two cents: _subspaces_ is good, especially if there is a conceptual link to an underlying Hilbert space. If we are referring to subsets of qubits, we can simply use _subsets_.

Reply:
Sounds good! I went back and forth on using _subgroup_ because of this. It's helpful to know it sticks out to others--definitely confusing now that I think about it. It seems like _subsets_ would be the right choice here because the connection to _subspaces_ is not crucial in this context (maybe more for "What is the Theory behind REM"). I'll change it to _subsets_ for now, but if _subspaces_ seems better we can revise to that.